### DISCUSSION TITLE
How to declare controls for each story

### DISCUSSION THREAD

**Discussion by amrabuaza** [jonspalmer/view_component_storybook#94] [Q&A]
I am using rails engine that's include view components and storybook , 
Inside test directory i am using view component stories but i can not specify  each story of his controls
I'am using view_component_storybook version : 0.8.0 
because I'am using ruby version 2.7.2

```
class TextInputComponentStories < ViewComponent::Storybook::Stories
  story(:input) {}
  story(:with_postfix) {}
  story(:with_prefix) {}
  story(:with_helper_text) {}
  story(:disabled) {}
end
```

So here i need to specify controls for input story
I'm try using 

```
 story :input do
    controls do
      text(:object, 'employee')
    end
  end
```
But dose not work when try to write json 
It's return 'ActiveModel::ValidationError: Validation failed: Param is not included in the list'

And i can not use constructor because view component storybook version dos't support constructor

**Comment by jonspalmer** [comment]
The Validation error that normally means that the name of the control you're using doesn't line up with a constructor parameter of the ViewComponent that you're instantiating. The Validation errors are much stronger in the latest versions too. Perhaps you could share your ViewComponent code for TextInputComponent?

Perhaps more importantly you should be able to use the latest version of the Gem on 2.7.x. We test it against 2.7.4 but I'm would be surprised if it didn't work won 2.7.2.

**Reply by amrabuaza** [reply]
Hello there
This the text input component code

```ruby

class TextInputComponent < ViewComponent::Base
  attr_accessor :options

  BASE_CLASSES = %w[input-component form-control].freeze

  def initialize(options = {})
    @options = options
  end

  def classes
    (options[:class].to_a + BASE_CLASSES).join(' ')
  end
end
```

Thank you

**Reply by jonspalmer** [reply]
Your VC expects a positional argument `options`. We don't support positional arguments until the later versions that use the `constructor` declaration. Alternatively you could make `options` a named parameter of your VC and then change story to match (ie `options` instead of `object`. 

I really think you should just upgrade to a newer version of this gem.
